noir-projects/aztec-nr/tests/src/note_getter_test.nr

```diff
@@ -0,0 +1,124 @@
+use dep::protocol_types::constants::MAX_NOTE_HASH_READ_REQUESTS_PER_CALL;
+use dep::aztec::context::PrivateContext;
+use dep::aztec::note::{
+    note_header::NoteHeader,
+    note_getter_options::{NoteGetterOptions, Sort, SortOrder, Comparator, PropertySelector},
+    note_getter::_get_notes_constrain_get_notes_internal,
+};
+use dep::aztec::protocol_types::address::AztecAddress;
+use crate::mock::test_note::TestNote;
+
+#[test]
+fn sets_note_manually_and_fetches_it() {
+    let mut context: PrivateContext = dep::std::unsafe::zeroed();
+    context.inputs.call_context.storage_contract_address = AztecAddress::from_field(69);
+
+    let mut test_note = TestNote::new(1337);
+    test_note.header = NoteHeader::new(AztecAddress::from_field(69), 0, 42);
+
+    let mut opt_notes: [Option<TestNote>; MAX_NOTE_HASH_READ_REQUESTS_PER_CALL] = [Option::none(); MAX_NOTE_HASH_READ_REQUESTS_PER_CALL];
```

Comment:
```suggestion
    let mut opt_notes = [Option::none(); MAX_NOTE_HASH_READ_REQUESTS_PER_CALL];
```

Just a heads up. I think you don't need to specify the type here and Noir should derive it on its own.